src/types/timestamp.rs

```diff
@@ -413,66 +409,80 @@ impl Timestamp {
-                // is `9500` (only 4 digits), we need to add two leading zeros to make: `009500`.
-                let actual_num_digits = super::num_decimal_digits_in_u64(scaled as u64);
-                let num_leading_zeros = (*num_digits as u64) - actual_num_digits;
-                write!(output, ".")?;
-                for _ in 0..num_leading_zeros {
-                    write!(output, "0")?;
-                }
-                write!(output, "{}", scaled)?;
-                Ok(())
-            }
-            Mantissa::Arbitrary(decimal) => {
-                let exponent = decimal.exponent;
-                let coefficient = &decimal.coefficient;
-                if exponent >= 0 {
-                    // We know that the coefficient is non-zero (the mantissa was not empty),
-                    // so having a positive exponent would result in an illegal fractional
-                    // seconds value.
-                    panic!("found fractional seconds decimal that was >= 1.");
-                }
 
-                let num_digits = decimal.coefficient.number_of_decimal_digits();
-                let abs_exponent = decimal.exponent.unsigned_abs();
-                // At this point, we know that the abs_exponent is greater than num_digits because
-                // the decimal has to be < 1.
-                let num_leading_zeros = abs_exponent - num_digits;
-                write!(output, ".")?;
-                for _ in 0..num_leading_zeros {
-                    write!(output, "0")?;
-                }
-                if coefficient.is_negative_zero() {
-                    write!(output, "0")?;
-                } else if coefficient.sign == Negative {
-                    panic!("cannot have a negative coefficient (other than -0)");
-                } else {
-                    write!(output, "{}", decimal.coefficient)?;
-                }
-                Ok(())
-            }
+        write!(output, "-{:0>2}", datetime.day())?;
+        //                   ^-- delimiting hyphen and 0-padded, right aligned, 2-digit day
+        if self.precision == Precision::Day {
+            write!(output, "T")?;
+            return Ok(());
+        }
+
+        write!(
+            output,
+            "T{:0>2}:{:0>2}",
+            // ^-- delimiting T, formatted hour, delimiting colon, formatted minute
+            datetime.hour(),
+            datetime.minute()
+        )?;
+        if self.precision == Precision::HourAndMinute {
+            self.format_offset(offset_minutes, output)?;
+            return Ok(());
         }
+
+        write!(output, ":{:0>2}", datetime.second())?;
+        //                   ^-- delimiting colon, formatted second
+        self.format_fractional_seconds(output)?;
+        self.format_offset(offset_minutes, output)?;
+        Ok(())
+    }
+
+    fn format_offset<W: std::fmt::Write>(
+        &self,
+        offset_minutes: Option<i32>,
+        output: &mut W,
+    ) -> IonResult<()> {
```

Comment:
We can from this function... but it's called by `write (soon to be format)_timestamp` and for that, I think we want it to stay `IonResult` because that is used by the `text_formatter` (which does have some error conditions). Check me on that tho!

Reply:
I think we can remove the `IonResult` from here and even from `write_offset` defined for `write_timestamp` and `format_offset` defined for `text_formatter`. 
Here's how its used in the `text_formatter`(similar usage in `write_timestamp`):
```rust
if value.precision == Precision::HourAndMinute {
            self.format_offset(offset_minutes)?;
            return Ok(());
        }
```
It just uses propagates the error from `write_offset`. But as `write_offset` doesn't have any errors, I think we can remove wrapping with `IonResult` here and then again unwrapping at `text_formatter` or `write_timestamp`.
That being said we can certainly do it in a separate PR including change for `text_formatter` and `write_timestamp`.

Reply:
Ok. In offline chat we came to agreement that there are some smaller possible tweaks within the scope of this PR, but they don't necessitate changes now.